server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestUpgradeActionDeprecated.java

```diff
@@ -32,11 +34,19 @@
 
 import static org.elasticsearch.rest.RestRequest.Method.POST;
 
-public class RestUpgradeAction extends BaseRestHandler {
+public class RestUpgradeActionDeprecated extends BaseRestHandler {
```

Comment:
I don't think there's any reason to rename the class, we can mark it with `@Deprecated` which would be a better indicator.

Reply:
I checked how `controller.registerAsDeprecatedHandler` is used and found that all handlers that use has _Deprecated_ suffix in class name (e.g., `RestDeleteTransformActionDeprecated`). I  think this is a name convention so I renamed classname here too. Should I return old name back in spite of this?

Reply:
Looks like you are correct, thanks for catching that! This looks good then.